File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int>res;
        unordered_map<int,int>mp;

        for(auto num : nums){
            mp[num]++;
        }
        for(auto [key,val] : mp){
            if(val >1){
                res.push_back(key);
            }
        }

        return res;
        
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        
        List<Integer>ans = new ArrayList<>();

        Map<Integer,Integer>mp = new HashMap<>();

        for(int num : nums){
            mp.put(num, mp.getOrDefault(num, 0) + 1);

        }

        for(Map.Entry<Integer,Integer> entry : mp.entrySet()){
            int k = entry.getKey();
            int v = entry.getValue();

            if(v > 1){
                ans.add(k);
            }

        }

        return ans;
    }
}